src/cloudflare/internal/workflows-api.ts

```diff
@@ -0,0 +1,10 @@
+export class NonRetryableError extends Error {
+  // `__brand` is needed for Workflows' engine to validate if the user returned a NonRetryableError
+  // this provides better DX because they can just extend NonRetryableError for their own Errors
+  public readonly __brand: string = 'NonRetryableError';
```

Comment:
Non-blocking but...This is a bit unfortunate. I'd almost prefer something like `NonRetryableError.isInstance(err)` that can check an internal private... e.g.

```js
const kTag = Symbol('kTag');

class NonRetryableError extends Error {
  [kTag] = kTag;  

  static isInstance(value) { return value[kTag] === kTag; }
}
```

Reply:
Is your suggested implementation serialized by RPC? If yes, I'm fine with replacing it 😅

Reply:
ah, no, it would not be. Would you mind adding a comment then to indicate that the field is a public own property because it needs to be serialized for RPC?

Reply:
Why don't we use name property as a way if distinguishing this error type?